tests/test_utils.py

```diff
@@ -694,4 +700,95 @@ def test_get_charm_name_from_path_wrong_name(tmp_path, build_charm_directory, na
     assert exc_info.value.args[0] == f"Path does not contain source for a valid charm: {full_path}"
 
 
+# endregion
+# region Tests for pip-related functions.
+
+
+@pytest.mark.parametrize(
+    ("requirements", "expected"),
+    [
+        pytest.param([], set(), id="empty"),
+        pytest.param(["abc==1.0.0"], {"abc==1.0.0"}, id="simple"),
+        pytest.param(["-e ."], set(), id="editable-ignored"),
+    ],
+)
+def test_get_pypi_packages(requirements, expected):
+    assert get_pypi_packages(requirements) == expected
+
+
+@pytest.mark.parametrize(("packages", "expected"), [({"abc"}, {"abc"}), ({"abc==1.0.0"}, {"abc"})])
+def test_get_package_names(packages, expected):
+    assert get_package_names(packages) == expected
```

Comment:
We may want to test all operators: `~=`, `==`, `!=`, `>=`, `<=`, `>`, `<`, `===`

Reply:
Agreed, thanks! Updated.